docs/src/design/multiset.md

```diff
@@ -1,56 +1,158 @@
+The first is achieved using [virtual tables](#virtual-tables) of data, where we add a row at some cycle in the trace and remove it at a later cycle when it is needed again. Instead of maintaining the entire table in the execution trace, multiset checks allow us to prove data consistency of this table using one running product column.
 
-1. allows us to enforce constraints and prove the correctness of the result for the given inputs.
-2. allows us to do this more efficiently then if we actually had to execute this without a specialized design.
+The second is done by reducing each operation to a lookup value and then using a [communication bus](#communication-buses), implemented as a running product column, to provably connect the two sections of the trace.
 
-The co-processor takes in the two inputs and the bitwise operation and then provably computes the result so we could just look it up, which is exactly what we need(More information on [bitwise design doc](./chiplets/bitwise.md)).
+### Running product columns
+Although the multiset equality check can be thought of as comparing multiset equality between two vectors $a$ and $b$, in Miden VM it is implemented as a single running product column in the following way:
 
-But now we’re left with a problem - we can prove that the bitwise computation is done correctly, but it’s not connected to our main processor yet, so we still need a way to actually “look up” the value and prove that the value we’re using after our lookup is in fact the correct result from that co-processor for the inputs we specified.
-
-**NOTE** - The stack and the lookup value might not necessarily be sharing the same trace. In fact, the likelihood of both sharing the same/adjacent row is very low. Given STARK constraints are local(constraints can be applied only on two consecutive traces), we can't enforce any transition constraint here.
-
-**Permutation checks are what allow us to link these different processors and prove that this lookup is being executed correctly.**
+- The running product column is initialized to $1$ at the beginning of the trace.
+- All values of $a$ are multiplied into the running product column.
+- All values of $b$ are divided out of the running product column.
+- If $a$ and $b$ were multiset equal, then the running product column will equal $1$ at the end of the trace.
 
 ### Cost of running product columns
-It is important to note that depending on the field in which we operate, a running product column may actually require more than one trace columns. This is specifically true for small field.
+It is important to note that depending on the field in which we operate, a running product column may actually require more than one trace column. This is specifically true for small field.
 
 For example, if we are in a 64-bit field, each running product column would need to be represented by $2$ columns to achieve ~100-bit security, and by $3$ columns to achieve ~128-bit security.
```

Comment:
I'd probably make this section much more miden specific. Meaning, we know we are in a 64-bit field - so, no need to say "if we are in a 64-bit filed" etc.

Reply:
Ah - actually I just kept the section that you had written previously. I could change it to:

"Since Miden uses a 64-bit field, each running product column needs to be represented by $2$ columns to achieve ~100-bit security and by $3$ columns to achieve ~128-bit security."

Is that sufficient or do you want more adjustment?

Reply:
Yep - that works!